### ISSUE TITLE
Representer support for JSONAPI and other formats.

### ISSUE THREAD

**Issue by undergroundwebdesigns** [apotonick/trailblazer#76]
The Representer class currently contains the line `# TODO: so far, we only support JSON, but it's two lines to change to support any kind of format.`. What needs to be changed? I just tried using JSON API and did hit errors, but it's not immediately clear to me what needs to be changed to use other Responder formats with Trailblazer. I'd be happy to submit a PR if you can give me an idea of where to start?

**Comment by apotonick** [comment]
So far, you just need to include the appropriate representer modules, the only public API is `Operation#to_json` which is called from the responder, so you have to make sure this gets called for whatever format you use.

Parsing-wise we (luckily) handle all that _in_ the operation. Make sure to use TRB master! Oh, and I describe all that in chapter 11 but I use HAL as I dislike JSON-API.

**Comment by undergroundwebdesigns** [comment]
Ah, my mistake. I was just using it wrong. Specifically, if you don't set a type and/or don't have a corresponding root key in the json you're feeding it, the JSON-API representer will give a weird `NoMethodError: undefined method `[]' for nil:NilClass` error that can be confusing.

tl;dr: Include a type in your representer and use well-formed JSON-API and it works :)
